source/App.cpp

```diff
@@ -1543,28 +1585,31 @@ void App::onUserInput(UserInput* ui) {
 					if (sess->canFire()) {
 						fired = true;
 						sess->countClick();														// Count clicks
-						Array<shared_ptr<Entity>> dontHit;
-						dontHit.append(m_currentDecals);
+						Array<shared_ptr<Entity>> dontHit = { m_hitDecal };
+						dontHit.append(m_currentMissDecals);
 						dontHit.append(m_explosions);
 						for (auto projectile : m_projectileArray) { dontHit.append(projectile.entity); }
 						Model::HitInfo info;
 						float hitDist = finf();
 						int hitIdx = -1;
+
 						shared_ptr<TargetEntity> target = m_weapon->fire(targetArray, hitIdx, hitDist, info, dontHit);			// Fire the weapon
+
+						WeaponConfig& wConfig = sessConfig->weapon;
 						if (notNull(target)) {					// Check if we hit anything
 							hitTarget(target);					// If we did, we are in hitscan mode, apply the damage and manage the target here
+
+							const Vector3& camDir = -activeCamera()->frame().lookVector();
+							drawDecal(info.point + 0.01f*camDir, camDir, true);
 						}
 						else {
 							if(!sessConfig->weapon.isLaser()) { 
 								m_sceneHitSound->play(sessConfig->audio.sceneHitSoundVol); 
 							}
-							WeaponConfig& wConfig = sessConfig->weapon;
 							// Draw a decal here if we are in hitscan mode
-							if (wConfig.hitScan && wConfig.renderDecals && hitDist < finf()) {
-								CFrame frame = activeCamera()->frame();
+							if (wConfig.hitScan && hitDist < finf()) {
 								// Draw decal at the lookRay/world intersection
-								Point3 position = frame.translation + frame.lookRay().direction() * (hitDist - 0.01f);
-								drawDecal(position, info.normal);
+								drawDecal(info.point + 0.01f*info.normal, info.normal);
```

Comment:
The old code was a hack without comment. Might want to add a comment here.

Reply:
This is done 4 places throughout `App.cpp`. Does it make sense to just add a comment to each location this is done? 

Does something along the lines of:
```
// Offset slightly along normal to avoid Z-fighting the wall
```
make sense?

Reply:
In case the "+1" wasn't clear enough, I support the use of this comment throughout.

Reply:
@jspjutNV comments have been added. Feel free to resolve this if they are sufficient.